### ISSUE TITLE
Streamline2 branch failing.

### ISSUE THREAD

**Issue by rgleason** [rgleason/testplugin_pi#55]
https://github.com/rgleason/testplugin_pi/tree/streamline   <---NOW Using streamline2   This one fails.

Need to package the json-validate stuff so it is completely separate from the Testplugin ocpnsrc and ocpninclude directories so that this can easily be used in other plugins..

Right now I am getting failures to compile due to

```
  tpJSON.cpp
C:\Users\fcgle\source\testplugin_pi\src\tpJSON.cpp(122,17): warning C4101: 'l_BoundaryState': unreferenced local variable [C:\Users\fcgle\source\testplugin_pi\build\testplugin_pi.vcxproj]
C:\Users\fcgle\source\testplugin_pi\src\tpJSON.cpp(117,17): warning C4101: 'l_dCrossingLat': unreferenced local variable [C:\Users\fcgle\source\testplugin_pi\build\testplugin_pi.vcxproj]
C:\Users\fcgle\source\testplugin_pi\src\tpJSON.cpp(112,17): warning C4101: 'l_dLon': unreferenced local variable [C:\Users\fcgle\source\testplugin_pi\build\testplugin_pi.vcxproj]
```

**Comment by rgleason** [comment]
@bdbcat   

Have managed to get the [testplugin/streamline2 branch ](https://github.com/rgleason/testplugin_pi/tree/streamline2) building, but it has some flakiness when running in OpenCPN, I've left comments in CMakeLists.txt.  I was trying to package json-schema-validator under libs.  It certainly can be simplified more.
Also this output.txt file shows a bunch of warnings because I haven't got something exactly right. 
[output.txt](https://github.com/rgleason/testplugin_pi/files/12528779/output.txt)

This took two tries, the other one is in the "streamline" which fails badly.

To see what has been done to the files, look at the top section.
[File-summary-revised.txt](https://github.com/rgleason/testplugin_pi/files/12528739/File-summary-revised.txt)

**Comment by rgleason** [comment]
https://github.com/rgleason/testplugin_pi/commit/bffe3ff7f577de2e17dd7f6b74a295488aafb3b7

**Comment by rgleason** [comment]
Have reached an impass trying to simplify and package json validation.
I can't get it to build locally with Windows.
This is the same failure as circleci msvc-wx32 
https://app.circleci.com/pipelines/github/rgleason/testplugin_pi/366/workflows/92e178c5-70e4-423b-89ee-2c11bcddf35d/jobs/3987

First cleaned up and used libs/ from weather_routing. 
Then worked on CmakeLists.txt format to bring the plugins into alignment.
Then reviewed json-schema-validator and files and started working on it.
Use Testplugin Streamline2 Branch
https://github.com/rgleason/testplugin_pi/commits/streamline2

Other OS are building https://cloudsmith.io/~opencpn/repos/testplugin-alpha/packages/?q=1.0.254.12
But these are probably not working right.

**Comment by rgleason** [comment]
Still trying to untangle this. The validator could possibly be used in watchdog, weather_routing and ocpn_draw because they all use ODAPI for messaging, so I think it would be better if the json-validation was packaged somehow and in Libs.

Have pushed v1.0.254.13 up to Streamline2 Branch.
1. Have updated libs with the new packages.
2. Commented out validation in cmakelists.txt to see if I could get it building.
3. Found that I could not.
4. I found that TPjson.cpp has references to files that have changed and I don't know how to redirect.
5. The output.txt file is attached.

[output.txt](https://github.com/rgleason/testplugin_pi/files/12554836/output.txt)
